src/main/java/mate/academy/bookshop/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+public class OrderServiceImpl implements OrderService {
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+
+    @Override
+    @Transactional
+    public OrderResponseDto createOrderByUser(User user, OrderRequestDto orderRequestDto) {
+        ShoppingCart shoppingCart = shoppingCartRepository
+                .findByUserId(user.getId())
+                .orElseThrow(() -> new EntityNotFoundException("Shopping cart not found"
+                        + " for user with id: " + user.getId()
+                ));
+        if (shoppingCart.getCartItems().isEmpty()) {
+            throw new EntityNotFoundException("Shopping cart is empty "
+                    + "for user id: " + user.getId());
+        }
+        Order order = orderMapper.toEntity(orderRequestDto);
+        order.setUser(user);
+        order.setStatus(Order.Status.PENDING);
+        order.setOrderDate(LocalDateTime.now());
+
+        Set<OrderItem> orderItemSet = shoppingCart
+                .getCartItems()
+                .stream()
+                .map(cart -> mapToOrderItem(cart, order))
+                .collect(Collectors.toSet());
+
+        order.setOrderItems(orderItemSet);
+
+        BigDecimal totalPrice = orderItemSet
+                .stream()
+                .map(OrderItem::getPrice)
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+
+        order.setTotal(totalPrice);
+        orderRepository.save(order);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public List<OrderResponseDto> getOrderByUser(User user) {
+        List<Order> ordersByUser = orderRepository.findAllByUser(user);
+        return orderMapper.toDto(ordersByUser);
+    }
+
+    @Override
+    @Transactional
+    public OrderResponseDto updateStatusByOrderId(Long orderId,
+                                                  OrderUpdateStatusDto orderUpdateStatusDto) {
+        Order order = orderRepository
+                .findById(orderId)
+                .orElseThrow(() -> new EntityNotFoundException("Order not found with id: "
+                        + orderId
+                ));
+        orderMapper.updateOrderStatusFromDto(orderUpdateStatusDto, order);
+        orderRepository.save(order);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    @Transactional
+    public List<OrderItemResponseDto> getOrderItemsByOrderId(Long orderId, Long userId) {
+        Order order = orderRepository
+                .findByIdAndUserId(orderId, userId)
+                .orElseThrow(() -> new EntityNotFoundException("Order items not found "
+                        + "for order id: " + orderId + " and user id: " + userId
+                ));
+        return orderItemMapper.toDto(orderItemRepository.findAllByOrder(order));
```

Comment:
create `List<OrderItemDto> toOrderItemDtoList(List<OrderItem> orderItems);` in `OrderItemMapper`

```suggestion
        return orderItemMapper.toOrderItemDtoList(order.getOrderItems());
```